tools/generator/Tests/Unit-Tests/ManagedTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Android.Runtime;
+using Mono.Cecil;
+using MonoDroid.Generation;
+using NUnit.Framework;
+using System.IO;
+using System.Linq;
+
+namespace Com.Mypackage
+{
+	[Register ("com/mypackage/foo")]
+	public class Foo
+	{
+		[Register ("foo", "()V", "")]
+		public Foo () { }
+
+		[Register ("bar", "()V", "")]
+		public void Bar () { }
+
+		[Register ("barWithParams", "(ZID)V", "")]
+		public void BarWithParams (bool a, int b, double c) { }
+
+		[Register ("value")]
+		public const int Value = 1234;
+	}
+
+	[Register ("com/mypackage/service")]
+	public interface IService { }
+}
+
+namespace generatortests
+{
+	[TestFixture]
+	public class ManagedTests
+	{
+		string tempFile;
+		ModuleDefinition module;
+
+		[SetUp]
+		public void SetUp ()
+		{
+			tempFile = Path.GetTempFileName ();
+			File.Copy (GetType ().Assembly.Location, tempFile, true);
+			module = ModuleDefinition.ReadModule (tempFile);
+		}
+
+		[TearDown]
+		public void TearDown ()
+		{
+			module.Dispose ();
+			if (File.Exists (tempFile))
+				File.Delete (tempFile);
+		}
+
+		[Test]
+		public void Class ()
+		{
+			var @class = new ManagedClassGen (module.GetType ("Com.Mypackage.Foo"));
+			Assert.AreEqual ("public", @class.Visibility);
+			Assert.AreEqual ("Foo", @class.Name);
+			Assert.AreEqual ("com.mypackage.foo", @class.JavaName);
+			Assert.AreEqual ("Lcom/mypackage/foo;", @class.JniName);
+			Assert.IsFalse (@class.IsAbstract);
+			Assert.IsFalse (@class.IsFinal);
+			Assert.IsFalse (@class.IsDeprecated);
+			Assert.IsNull (@class.DeprecatedComment);
+		}
+
+		[Test]
+		public void Method ()
+		{
+			var type = module.GetType ("Com.Mypackage.Foo");
+			var @class = new ManagedClassGen (type);
+			var method = new ManagedMethod (@class, type.Methods.First (m => m.Name == "Bar"));
+			method.FillReturnType ();
```

Comment:
I don't understand why this method call must be present: the `ManagedMethod` (and `XmlMethod`!) constructors *also* call `Method.FillReturnType()`. Thus, isn't this call superfluous and unnecessary?